vsts/scripts/pullAndTag.sh

```diff
@@ -9,23 +9,42 @@ set -euo pipefail
 # $2 > runtime-images-acr.txt
 declare imagefilter="oryxdevmcr.azurecr.io/public/oryx"
 
-echo "Build image filter is set"
-while read buildImage; do
-  # Always use specific build number based tag and then use the same tag to create a 'latest' tag and push it
-  if [[ $buildImage != *:latest ]]; then
-	echo "Pulling the build image $buildImage ..."
-	docker pull "$buildImage" | sed 's/^/     /'
+function tagBuildImageForIntegrationTest() {
+	local devbuildImageName="$1"
+	local devbuildImageType="$2"
+	
+	# Always use specific build number based tag and then use the same tag to create a 'latest' tag and push it
+	if [ -z "$devbuildImageType" ]; then
+		buildImage=$devbuildImageName:$buildDefName.$buildNumber
+		# Trim the build number tag and append the '':latest' to end of it
+		newtag=$devbuildImageName:latest
+	else
+		buildImage=$devbuildImageName:$devbuildImageType-$buildDefName.$buildNumber
+		newtag=$devbuildImageName:$devbuildImageType-latest
+	fi
 
-	# Trim the build number tag and append the '':latest' to end of it
-	newtag="${buildImage%:*}:latest"
+	echo "Pulling the build image $buildImage ..."
+ 	docker pull "$buildImage" | sed 's/^/     /'
 
 	echo
 	echo "Tagging the source image with tag $newtag ..."
+	
 	docker tag "$buildImage" "$newtag" | sed 's/^/     /'
```

Comment:
this was the bug ..

Digest: sha256:41c086374419d88f440d54499702f7c3de435e5a2dfa885e1ac19e108039a7fe
     Status: Downloaded newer image for oryxdevmcr.azurecr.io/public/oryx/build:azfunc-jamstack-Oryx-CI.20200929.1
     oryxdevmcr.azurecr.io/public/oryx/build:azfunc-jamstack-Oryx-CI.20200929.1

Tagging the source image with tag oryxdevmcr.azurecr.io/public/oryx/build:latest ...